File: tests/fixtures/go-scaffold/src/services/account_service.go
package services

type AccountRepository interface {
    Save(id string) error
}

type AccountService struct {
    repo AccountRepository
}

func newAccountService(repo AccountRepository) *AccountService {
    return &AccountService{repo: repo}
}

func (s *AccountService) CreateAccount(owner string) (string, error) {
    if owner == "" {
        return "", nil
    }
    accountID := "acct-123"
    return accountID, s.repo.Save(accountID)
}

func CalculateMonthlyAccounts() int {
    return 0
}

File: languages/javascript/tests/fixtures/ruby-scaffold/app/services/account_service.rb
module Services
  class AccountService
    def initialize(repository:)
      @repository = repository
    end

    def create_account(owner:)
      return nil if owner.nil? || owner.empty?

      account_id = "acct-#{owner.downcase}"
      repository.save(account_id)
      account_id
    end

    def self.build(repository:)
      new(repository: repository)
    end

    private

    attr_reader :repository
  end
end

def calculate_monthly_accounts
  0
end
